src/js/providers/video-listener-mixin.js

```diff
@@ -163,18 +164,34 @@ const VideoListenerMixin = {
         }
     },
     error() {
-        const code = (this.video.error && this.video.error.code) || -1;
-        const message = ({
-            1: 'Unknown operation aborted',
-            2: 'Unknown network error',
-            3: 'Unknown decode error',
-            4: 'File could not be played'
-        }[code] || 'Unknown');
-        this.trigger(MEDIA_ERROR, {
-            code: code,
-            message: 'Error loading media: ' + message
-        });
+        const errorCode = (this.video.error && this.video.error.code) || -1;
```

Comment:
`|| 0` since we're pulling from an Array now.

I think a little switch call to a static function that triggers the error would  be a lot simpler and easier to read.